### PR TITLE
Adding postsubmit pod-utilities job

### PR DESCRIPTION
**Pull request by RagniGarg97**
post-submit job to build pod-utilities image.
https://github.com/ppc64le-cloud/test-infra/blob/master/images/pod-utilities/Dockerfile

### CODE REVIEW

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/135#" title="Author self-approved">RagniGarg97</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mkumatag** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mkumatag` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                - name: default-user
+                  user:
+                    token: ${token}
+                EOL
+                export KUBECONFIG=$(pwd)/kubeconfig
+                export utils=(clonerefs entrypoint initupload sidecar)
+                pushd images/pod-utilities
+                for i in ${utils[@]}
+                do
+                    kubectl build --push --registry-secret quay-powercloud-regcred --namespace image-builder --build-arg ARCH=amd64 --build-arg POD_UTILITY=$i -t quay.io/powercloud/$i:amd64-v20210126-f0baa5f3d9 -f Dockerfile ./
```
Comment (mkumatag) [inline config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml:64]: read the version from somewhere like the logic used for the bazel

File: config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                    image: quay.io/powercloud/$i:ppc64le-v20210126-f0baa5f3d9
+                    platform:
+                      architecture: ppc64le
+                      os: linux
+                  -
+                    image: quay.io/powercloud/$i:amd64-v20210126-f0baa5f3d9
+                    platform:
+                      architecture: amd64
+                      os: linux
+                EOL
```
Comment (mkumatag) [inline config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml:94]: as mentioned above use the version from a file

File: config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                    token: ${token}
+                EOL
+                export KUBECONFIG=$(pwd)/kubeconfig
+                export utils=(clonerefs entrypoint initupload sidecar)
+                pushd images/pod-utilities
+                for i in ${utils[@]}
+                do
+                    kubectl build --push --registry-secret quay-powercloud-regcred --namespace image-builder --build-arg ARCH=amd64 --build-arg POD_UTILITY=$i -t quay.io/powercloud/$i:amd64-v20210126-f0baa5f3d9 -f Dockerfile ./
+                done
+                popd
```
Comment (mkumatag) [inline config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml:66]: we aren't building the images for the amd64 arch, instead we are pulling them, retagging, and pushing into our repo, not sure if this logic really works!

Comment (mkumatag) [reply config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml:66]: /cc @Rajalakshmi-Girish

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml (hunk 3 above)
Comment (Rajalakshmi-Girish) [reply config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml:66]: Yes as per the README available in pod-utilities folder, I have been only pulling, retagging, and pushing into our repo for amd64 arch.
I did not try building for amd64.

Comment (mkumatag) [reply config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml:66]: If we can build it for amd64 then I don't mind build for amd64 platform as well.

File: config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                do
+                    kubectl build --push --registry-secret quay-powercloud-regcred --namespace image-builder --build-arg ARCH=amd64 --build-arg POD_UTILITY=$i -t quay.io/powercloud/$i:amd64-$version -f Dockerfile ./
+                done
+                popd
+                export KUBECONFIG=/etc/kubeconfig/config
+                pushd images/pod-utilities
+                for i in ${utils[@]}
+                do
+                      kubectl build --push --registry-secret quay-powercloud-regcred --namespace image-builder --build-arg ARCH=ppc64le --build-arg POD_UTILITY=$i -t quay.io/powercloud/$i:ppc64le-$version -f Dockerfile ./
+                done
```
Comment (mkumatag) [inline config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml:73]: combine these 2 loops into one and use `--kubeconfig` flag for passing the kubeconfig file to ppc build

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                    token: ${token}
+                EOL
+                export KUBECONFIG=$(pwd)/kubeconfig
+                export utils=(clonerefs entrypoint initupload sidecar)
+                pushd images/pod-utilities
+                export version=`cat version.txt`
+                for i in ${utils[@]}
+                do
+                    kubectl build --push --registry-secret quay-powercloud-regcred --namespace image-builder --build-arg ARCH=amd64 --build-arg POD_UTILITY=$i -t quay.io/powercloud/$i:amd64-$version -f Dockerfile ./
+                    kubectl build --push --registry-secret quay-powercloud-regcred --kubeconfig /etc/kubeconfig/config --namespace image-builder --build-arg ARCH=ppc64le --build-arg POD_UTILITY=$i -t quay.io/powercloud/$i:ppc64le-$version -f Dockerfile ./
```
Comment (mkumatag) [inline config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml:66]: This isn't required:
```
--build-arg ARCH=ppc64le
```

I don't see any such arg in the dockerfile

**Review by mkumatag — CHANGES_REQUESTED** [review]

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/135#pullrequestreview-593229727" title="Approved">mkumatag</a>*, *<a href="https://github.com/ppc64le-cloud/test-infra/pull/135#" title="Author self-approved">RagniGarg97</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)~~ [mkumatag]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ltccci** [comment]
@RagniGarg97: Updated the `job-config` configmap in namespace `prow` at cluster `default` using the following files:
 - key `podutilities-postsubmit.yaml` using file `config/jobs/ppc64le-cloud/pod-utilities/podutilities-postsubmit.yaml`

<details>

In response to [this](https://github.com/ppc64le-cloud/test-infra/pull/135):

>post-submit job to build pod-utilities image.
>https://github.com/ppc64le-cloud/test-infra/blob/master/images/pod-utilities/Dockerfile


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**